bluesky/run_engine.py

```diff
@@ -2194,7 +2194,8 @@ def _status_object_completed(self, ret, p_event, pardon_failures):
             # TODO: need a better channel to move this information back
             # to the run task.
             with self._state_lock:
-                self._exception = FailedStatus(ret)
```

Comment:
A nicer version of this...
```python
try:
    raise FailedStatus(ret) from ret.exception(timeout=0)
except Exception as e:
    self._exception = e
```

Reply:
Have added this and made sure to propagate the traceback also